consensus/notifications.go

```diff
@@ -0,0 +1,84 @@
+package consensus
+
+// A block is composed of many transactions. Blocks that have transactions that
+// depend on other transactions, but the transactions must all be applied in a
+// deterministic order. Transactions cannot have inter-dependencies, meaning
+// that an output cannot be created and then spent in the same transaction. As
+// far as diffs are concenred, this means that the elements of a transaction
+// diff should be able to be applied in any order and the outcome should be the
+// same. The elements of a block diff however must be applied in a specific
+// order, as transactions may depend on each other.
```

Comment:
I'm confused as to why this comment is up here. It seems like it should be split up and moved to the `TransactionDiff` and `BlockDiff` docstrings.